x/swaprouter/msg_server.go

```diff
@@ -0,0 +1,147 @@
+package swaprouter
+
+import (
+	"context"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v13/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v13/x/gamm/pool-models/stableswap"
+	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	"github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
+)
+
+type msgServer struct {
+	keeper *Keeper
+}
+
+var (
+	_ balancer.MsgServer = (*msgServer)(nil)
+	// TODO: enable after gamm refactor is ported from `concentrated-liqudity-main`
+	// _ stableswap.MsgCreatorServer = (*msgServer)(nil)
+)
+
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+func NewBalancerMsgServerImpl(keeper *Keeper) balancer.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+// TODO: enable after gamm refactor is ported from `concentrated-liqudity-main`
+// func NewStableswapMsgServerImpl(keeper *Keeper) stableswap.MsgCreatorServer {
+// 	return &msgServer{
+// 		keeper: keeper,
+// 	}
+// }
+
+// CreateBalancerPool is a create balancer pool message.
+func (server msgServer) CreateBalancerPool(goCtx context.Context, msg *balancer.MsgCreateBalancerPool) (*balancer.MsgCreateBalancerPoolResponse, error) {
+	poolId, err := server.CreatePool(goCtx, msg)
+	if err != nil {
+		return nil, err
+	}
+	return &balancer.MsgCreateBalancerPoolResponse{PoolID: poolId}, nil
+}
+
+func (server msgServer) CreateStableswapPool(goCtx context.Context, msg *stableswap.MsgCreateStableswapPool) (*stableswap.MsgCreateStableswapPoolResponse, error) {
+	poolId, err := server.CreatePool(goCtx, msg)
+	if err != nil {
+		return nil, err
+	}
+	return &stableswap.MsgCreateStableswapPoolResponse{PoolID: poolId}, nil
+}
+
+// func (server msgServer) StableSwapAdjustScalingFactors(goCtx context.Context, msg *stableswap.MsgStableSwapAdjustScalingFactors) (*stableswap.MsgStableSwapAdjustScalingFactorsResponse, error) {
```

Comment:
TODO: re-enable after gamm refactor is ported

Reply:
I think this actually needs to be removed, thanks for catching